### PR TITLE
Export profile data

### PR DESCRIPTION
**Pull request by atdotde**
This introduces a csv file that contains the data from
the structs defined in profile.c, in particular all
deco information computed for the dive profle (including
NDL, TTS, ceilings, surface GFs etc).

Signed-off-by: Robert C. Helling <helling@atdotde.de>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This allows to export all data that we use to compute the dive profile (including deco information, interpolation etc). This can be useful to debugging. There was a request for such a feature on [Scuba Board](https://www.scubaboard.com/community/threads/subsurface-4-7-6-and-subsurface-mobile-2-0-1-released.555591/page-2#post-8596314)  by somebody who wants to overlay a video with dive computer data.

The files created tend to be a bit heavy (I first wrote an xml version but that even was a few times bigger).

### CODE REVIEW

File: core/save-profiledata.c [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	if (same_string(filename, "-")) {
+		f = stdout;
+	} else {
+		f = subsurface_fopen(filename, "w");
+	}
+	if (f) {
+		flush_buffer(&buf, f);
+		error = fclose(f);
+	}
+	if (error)
```
Comment (bstoeger) [inline core/save-profiledata.c:192]: I'm not sure I understand the logic here:
If `subsurface_fopen()` fails, error stays 0 and therefore no error message is produced. Or am I missing something?

File: core/save-profiledata.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#define SAVE_PROFILE_DATA_H
+
+#include "dive.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int save_profiledata(const char *filename, bool selected_only);
+void save_profiles_buffer(struct membuffer *b, bool select_only);
```
Comment (bstoeger) [inline core/save-profiledata.h]: Is there any plan to use this outside of save-profiledata.c? If not, then there seems to be no reason to export it.

File: desktop-widgets/divelogexportdialog.ui [hunk 3]
```diff
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>507</width>
-    <height>398</height>
+    <height>423</height>
```
Comment (bstoeger) [inline desktop-widgets/divelogexportdialog.ui:10]: Tool-assisted change, I presume...

**Comment by bstoeger** [comment]
Looks good to me in general, only a few minor code-related comments.

File: core/save-profiledata.c (hunk 1 above)
Comment (atdotde) [reply core/save-profiledata.c:192]: Honestly, this was copy-pasted from save_dives_logic() in save_xml.c

Comment (atdotde) [reply core/save-profiledata.c:192]: Ah, no, I deleted too much.

File: core/save-profiledata.c [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+int save_profiledata(const char *filename, const bool select_only)
+{
+	struct membuffer buf = { 0 };
+	FILE *f;
+	int error = 0;
+
+
+	save_profiles_buffer(&buf, select_only);
```
Comment (dirkhh) [inline core/save-profiledata.c:180]: double empty line

File: core/save-profiledata.c [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+		put_format(b, "\n");
+
+		for (int i = 0; i < pi.nr; i++) {
+			put_pd(b, &pi.entry[i]);
+			put_format(b, "\n");
+		}
+
+		put_format(b, "\n");
+
+	}
```
Comment (dirkhh) [inline core/save-profiledata.c:171]: that's a lot of vertical white space

File: core/save-profiledata.c [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+		put_int(b, entry->pressure[c][0]);
+		put_int(b, entry->pressure[c][1]);
+	}
+	put_int(b, entry->temperature);
+	put_int(b, entry->depth);
+	put_int(b, entry->ceiling);
+	for (int i = 0; i < 16; i++) {
+		put_int(b, entry->ceilings[i]);
+	}
+	for (int i = 0; i < 16; i++) {
```
Comment (dirkhh) [inline core/save-profiledata.c]: both above and below - we usually do those one line blocks without `{ }`

**Review by dirkhh — APPROVED** [review]
Some overly picky white-space comments, otherwise looks good

**Comment by atdotde** [comment]
Removed excessive white space and single line braces which both were left-overs from the initial xml version.

**Comment by atdotde** [comment]
Rebased on master to make CHANGELOG happy.

**Review by dirkhh — APPROVED** [review]
let's merge

**Pull request merged**